### PR TITLE
Added google Authentication feature

### PR DESCRIPTION
**Pull request by likhita24**
As mentioned in the issue #16 , Added Google authentication feature using AllAuth library and also reformed the existing signup page by creating a separate create profile page.
Right now, the signup page looks like in the attached SS below

Screenshot (11)
and the create profile page

Screenshot (12)

### CODE REVIEW

File: accounts/forms.py [hunk 1]
```diff
@@ -83,11 +52,30 @@ def __init__(self, *args, **kwargs):
...
+
+    phone = forms.CharField(min_length=10, max_length=13, widget=forms.TextInput(attrs={'placeholder': ' '}))
+    gender = forms.ChoiceField(choices=UserProfile.GENDER_CHOICES, required=True,
+                               widget=forms.Select(attrs={'class': 'mdb-select'}))
+    state = forms.ChoiceField(choices=UserProfile.STATE_CHOICES, required=True,
+                              widget=forms.Select(attrs={'class': 'mdb-select'}))
+
+    accommodation_required = forms.ChoiceField(choices=UserProfile.ACCOMMODATION_CHOICES,
+                                               widget=forms.Select(attrs={'class': 'mdb-select'}), required=False)
+
```
Comment (harsh-not-haarsh) [inline accounts/forms.py:76]: @likhita24 
You didn't add any validators here, add validators for phone number and referral code

**Review by harsh-not-haarsh — CHANGES_REQUESTED** [review]

File: accounts/forms.py (hunk 1 above)
Comment (likhita24) [reply accounts/forms.py:100]: @harsh-not-haarsh 
Added Form validations for the create profile form

File: accounts/forms.py [hunk 2]
```diff
@@ -66,6 +33,41 @@ def clean_email(self):
...
+        super(RegisterForm, self).__init__(*args, **kwargs)
+        self.fields['email'].widget.attrs['icon_name'] = "fa fa-envelope"
+        self.fields['username'].widget.attrs['icon_name'] = "fa fa-id-card"
+        self.fields['first_name'].widget.attrs['icon_name'] = "fa fa-user"
+        self.fields['last_name'].widget.attrs['icon_name'] = "fa fa-user"
+        self.fields['password1'].widget.attrs['icon_name'] = "fa fa-lock"
+        self.fields['password2'].widget.attrs['icon_name'] = "fa fa-lock"
+
+
+class PasswordResetCaptchaForm(PasswordResetForm):
```
Comment (harsh-not-haarsh) [inline accounts/forms.py:46]: @likhita24 Maybe get rid of this right now?

File: accounts/forms.py [hunk 3]
```diff
@@ -6,40 +6,14 @@
...
-            attrs={'class': 'mdb-autocomplete', 'maxlength': '128', 'placeholder': ' '}),
-        required=True)
-    state = forms.ChoiceField(choices=UserProfile.STATE_CHOICES, required=True,
-                              widget=forms.Select(attrs={'class': 'mdb-select'}))
-
-    accommodation_required = forms.ChoiceField(choices=UserProfile.ACCOMMODATION_CHOICES,
-                                               widget=forms.Select(attrs={'class': 'mdb-select'}), required=False)
-    referred_by = forms.CharField(
-        max_length=8, required=False, widget=forms.TextInput(attrs={'placeholder': ' '}))
+    username = forms.CharField(required=False)
```
Comment (harsh-not-haarsh) [inline accounts/forms.py:9]: @likhita24 Don't hardcode fields

File: accounts/forms.py [hunk 4]
```diff
@@ -66,6 +33,41 @@ def clean_email(self):
...
+        self.fields['password2'].widget.attrs['icon_name'] = "fa fa-lock"
+
+
+class PasswordResetCaptchaForm(PasswordResetForm):
+    email = forms.EmailField(required=True)
+
+
+class CreateUserProfileForm(forms.ModelForm):
+
+    phone = forms.CharField(max_length=13)
```
Comment (harsh-not-haarsh) [inline accounts/forms.py:52]: @likhita24 Don't hardcode the fields, import them from model only

File: client_secret_1083763914956-0jlme6quju4qqdbq4rfjh3n21nhiob0n.apps.googleusercontent.com.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{"web":{"client_id":"1083763914956-0jlme6quju4qqdbq4rfjh3n21nhiob0n.apps.googleusercontent.com","project_id":"varchas","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"hYZ1GFmF_v0byQxwnZRAvYn3","redirect_uris":["http://127.0.0.1:8000/accounts/google/login/callback/"],"javascript_origins":["http://127.0.0.1:8000"]}}
```
Comment (harsh-not-haarsh) [inline client_secret_1083763914956-0jlme6quju4qqdbq4rfjh3n21nhiob0n.apps.googleusercontent.com.json:1]: shouldn't be a part of the repo

File: main/views.py [hunk 6]
```diff
@@ -14,13 +15,18 @@ class IndexView(TemplateView):
     template_name = 'main/index.html'
 
     def get_context_data(self, **kwargs):
-        if self.request.user.username != "":
-            userprofile = get_object_or_404(UserProfile, user=self.request.user)
+        user = self.request.user
+        if user.username != "":
```
Comment (harsh-not-haarsh) [inline main/views.py:19]: @likhita24 You can check user.is_authenticated()

File: accounts/views.py [hunk 7]
```diff
@@ -99,6 +97,49 @@ def joinTeam(request):
...
+    template_name = 'accounts/createprofile.html'
+
+    def form_valid(self, form):
+        user = self.request.user
+        if user.username != "":
+            data = self.request.POST.copy()
+            CreateUserProfileForm(data)
+            user.username = user.email
+            user.save()
+            profile = UserProfile(user=user, gender=data['gender'], phone=data['phone'],
```
Comment (harsh-not-haarsh) [inline accounts/views.py:121]: this is not the best way to do it, you can just save the form.

File: accounts/views.py [hunk 8]
```diff
@@ -99,6 +97,49 @@ def joinTeam(request):
...
+                                  college=data['college'], state=data['state'],
+                                  referral=data['referred_by'],
+                                  accommodation_required=data['accommodation_required'])
+            profile.save()
+            return HttpResponseRedirect(reverse('main:home'))
+        email = self.request.session['email']
```
Comment (harsh-not-haarsh) [inline accounts/views.py:127]: just get user from request

File: accounts/views.py [hunk 9]
```diff
@@ -99,6 +97,49 @@ def joinTeam(request):
     return render(request, 'accounts/joinTeam.html')
 
 
+def GoogleLogin(request):
```
Comment (harsh-not-haarsh) [inline accounts/views.py:100]: The code should be abstract, rename google login to just login

**Review by harsh-not-haarsh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**